tests/js/click/click-basic-tests.js

```diff
@@ -0,0 +1,143 @@
+                return gamepad.tests.utils.element.click(0);
+            };
+
+            // Initialize the webpage, i.e., focus on the button element.
+            $("button").focus();
+
+            // Confirm that the instance of the gamepad navigator is created.
+            gamepad.tests.navigator = gamepad.inputMapper({
+                windowObject: gamepad.tests.windowObject
+            });
+            jqUnit.assertTrue("The Gamepad Navigator should be instantiated.", fluid.isComponent(gamepad.tests.navigator));
+
+            // Check the state of gamepad inputs and webpage after polling.
+            gamepad.tests.navigator.pollGamepads();
+            jqUnit.assertLeftHand("The gamepad should be connected with no buttons/axes disturbed initially.", gamepad.tests.modelAtRest, gamepad.tests.navigator.model);
+            jqUnit.assertEquals("The focus should not have changed.", document.querySelector("button"), document.activeElement);
+
+            // Update the gamepad to click on the button element.
+            gamepad.tests.navigator.pollGamepads();
+
+            // Check if the button has been clicked.
+            jqUnit.assertTrue("The button should have been clicked.", gamepad.tests.clicked);
+
+            // Restore the gamepad back to its neutral state i.e., release the button.
+            gamepad.tests.navigator.pollGamepads();
+        });
+
+        jqUnit.test("Radio buttons can be selected by click.", function () {
+            gamepad.tests.windowObject.navigator.getGamepads = function () {
+                return gamepad.tests.utils.element.click(0);
+            };
+
+            // Initialize the webpage, i.e., focus on the radio button.
+            $("#radio-one").focus();
+
+            // Confirm that the instance of the gamepad navigator is created.
+            gamepad.tests.navigator = gamepad.inputMapper({
+                windowObject: gamepad.tests.windowObject
+            });
+            jqUnit.assertTrue("The Gamepad Navigator should be instantiated.", fluid.isComponent(gamepad.tests.navigator));
+
+            // Check the state of gamepad inputs and webpage after polling.
+            gamepad.tests.navigator.pollGamepads();
+            jqUnit.assertLeftHand("The gamepad should be connected with no buttons/axes disturbed initially.", gamepad.tests.modelAtRest, gamepad.tests.navigator.model);
+            jqUnit.assertEquals("The focus should not have changed.", document.querySelector("#radio-one"), document.activeElement);
+
+            // Update the gamepad to click on the radio button.
+            gamepad.tests.navigator.pollGamepads();
+
+            // Check if the radio button has been clicked.
+            jqUnit.assertTrue("The radio button should have been selected after clicking.", document.querySelector("#radio-one").checked);
+
+            // Restore the gamepad back to its neutral state i.e., release the button.
```

Comment:
I'm a little confused about how/whether the button is released by this call, you didn't change the list of gamepads.  Is this what the count variable you used does, limit the activity to one polling cycle?

Reply:
Each value of count is associated with one polling cycle, i.e., for each polling cycle, we have a new value of count that we changed (decreased) at the end of the previous polling cycle. So we set the value of the count variable before every test unit and then we know that in a particular polling cycle, we want the button to be pressed or be released (by using the value of count which we change in each polling cycle).

Reply:
Yes, I get that now.  I still think you can at least clean this up to have a variable that is local to the gamepad mock.